include/picongpu/plugins/xrayScattering/XrayScattering.hpp

```diff
@@ -0,0 +1,846 @@
+/* Copyright 2013-2020 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch,
+ *                     Klaus Steiniger, Felix Schmitt, Benjamin Worpitz,
+ *                     Juncheng E, Pawel Ordyna
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/particles/traits/SpeciesEligibleForSolver.hpp"
+#include "picongpu/plugins/ISimulationPlugin.hpp"
+#include "picongpu/plugins/common/stringHelpers.hpp"
+
+#include "picongpu/fields/FieldTmp.hpp"
+#include "picongpu/param/xrayScattering.param"
+#include "picongpu/plugins/xrayScattering/beam/XrayScatteringBeam.hpp"
+#include "picongpu/plugins/xrayScattering/XrayScattering.kernel"
+#include "picongpu/plugins/xrayScattering/XrayScatteringWriter.hpp"
```

Comment:
Could you check if there are circular include dependencies between this file, and each of those two ? From the first glance both pairs seem to have it.